src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/Extensions/Chef/SetAzureVMChefExtension.cs

```diff
@@ -66,6 +66,12 @@ public class SetAzureVMChefExtensionCommand : VirtualMachineChefExtensionCmdletB
         public string RunList { get; set; }
 
         [Parameter(
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "A JSON string to be added to the first run of chef-client.")]
+        [ValidateNotNullOrEmpty]
+        public string JsonAttributes { get; set; }
```

Comment:
In service management this.PublicConfiguration is string and I am not able to assign Hash to string. I am new to c#. I tried following but I am getting error 
` //var hashTable = new Hashtable();
            //hashTable.Add(BootstrapVersionTemplate, this.BootstrapVersion);
            //hashTable.Add(ClientRbTemplate, ClientConfig);

            //if (!IsRunListEmpty)
            //{
            //    hashTable.Add(RunListTemplate, this.RunList);
            //}

            //if (!IsBootstrapOptionsEmpty)
            //{
            //    hashTable.Add(BootStrapOptionsTemplate, this.BootstrapOptions);
            //}

            //if (!IsJsonAttributesEmpty)
            //{
            //    hashTable.Add(JsonAttributesTemplate, JsonAttributes);
            //}

            //if (!IsChefServiceIntervalEmpty)
            //{
            //    hashTable.Add(ChefServiceIntervalTemplate, this.ChefServiceInterval);
            //}

            //this.PublicConfiguration = hashTable.ToString();`